drivers/storage/portworx/deployment_test.go

```diff
@@ -1904,12 +1969,240 @@ func TestPodSpecForKvdbAuthErrorReadingSecret(t *testing.T) {
+
+	zoneToInstancesMap := map[string]int{"a": 3, "b": 3, "c": 2}
+	driver := portworx{
+		k8sClient:          k8sClient,
+		recorder:           record.NewFakeRecorder(0),
+		zoneToInstancesMap: zoneToInstancesMap,
+		cloudProvider:      "mock",
+	}
+
+	// Max storage nodes
+	maxNodes := uint32(3)
+	cluster.Spec.CloudStorage = &corev1alpha1.CloudStorageSpec{
+		MaxStorageNodes: &maxNodes,
+	}
+	expectedArgs := []string{
+		"-c", "px-cluster",
+		"-x", "kubernetes",
+		"-max_drive_set_count", "3",
+	}
+
+	// TC 1, no CloudStorage CapacitySpecs
+	actual, _ := driver.GetStoragePodSpec(cluster, nodeName)
+	list, err := driver.storageNodesList(cluster)
+	assert.NoError(t, err, "Error getting StorageNodeList")
+	assert.ElementsMatch(t, expectedArgs, actual.Containers[0].Args)
+	assert.Equal(t, 0, len(list), "no storage nodes are expected at this point")
+
+	// Storage devices and auto-compute max_storage_nodes_per_zone
+	cluster.Spec.CloudStorage = &corev1alpha1.CloudStorageSpec{
+		CapacitySpecs: []corev1alpha1.CloudStorageCapacitySpec{
+			{
+				MinIOPS:          uint32(100),
+				MinCapacityInGiB: uint64(100),
+				MaxCapacityInGiB: uint64(200),
+				Options:          map[string]string{"foo1": "bar1"},
+			},
+			{
+				MinIOPS:          uint32(200),
+				MinCapacityInGiB: uint64(200),
+				MaxCapacityInGiB: uint64(500),
+				Options:          map[string]string{"foo3": "bar3"},
+			},
+		},
+	}
+
+	expectedArgs = []string{
+		"-c", "px-cluster",
+		"-x", "kubernetes",
+		"-s", "type=foo,size=120,iops=110,foo1=bar1",
+		"-s", "type=bar,size=220,iops=210,foo3=bar3",
+		"-max_storage_nodes_per_zone", "2",
+	}
+
+	inputInstancesPerZone := 2
+	mockStorageManager.EXPECT().
+		GetStorageDistribution(&cloudops.StorageDistributionRequest{
+			ZoneCount:        len(zoneToInstancesMap),
+			InstancesPerZone: inputInstancesPerZone,
+			UserStorageSpec: []*cloudops.StorageSpec{
+				{
+					IOPS:        uint32(100),
+					MinCapacity: uint64(100),
+					MaxCapacity: uint64(200),
+				},
+				{
+					IOPS:        uint32(200),
+					MinCapacity: uint64(200),
+					MaxCapacity: uint64(500),
+				},
+			},
+		}).
+		Return(&cloudops.StorageDistributionResponse{
+			InstanceStorage: []*cloudops.StoragePoolSpec{
+				{
+					DriveCapacityGiB: uint64(120),
+					DriveType:        "foo",
+					DriveCount:       1,
+					InstancesPerZone: inputInstancesPerZone,
+					IOPS:             uint32(110),
+				},
+				{
+					DriveCapacityGiB: uint64(220),
+					DriveType:        "bar",
+					DriveCount:       1,
+					InstancesPerZone: inputInstancesPerZone,
+					IOPS:             uint32(210),
+				},
+			},
+		}, nil)
+
+	actual, err = driver.GetStoragePodSpec(cluster, nodeName)
+	assert.NoError(t, err, "Unexpected error on GetStoragePodSpec")
+	assert.ElementsMatch(t, expectedArgs, actual.Containers[0].Args)
+	assert.Equal(t, inputInstancesPerZone, cluster.Status.Storage.StorageInstancesPerZone, "wrong number of instances per zone in a the cluster status")
```

Comment:
We should do this for all below test cases, to ensure that the count hasn't changed in the status after different runs.